init-distro-rid.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+
+# initNonPortableDistroRid
+#
+# Input:
+#   buildOs: (str)
+#   buildArch: (str)
+#   isPortable: (int)
+#   rootfsDir: (str)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# initNonPortableDistroRid will attempt to initialize a non portable rid. These
+# rids are specific to distros need to build the product/package and consume
+# them on the same platform.
+#
+# If -portablebuild=false is passed a non-portable rid will be created for any
+# distro.
+#
+# Below is the list of current non-portable platforms.
+#
+# Builds from the following *must* be non-portable:
+#
+#   |    OS     |           Expected RID            |
+#   -------------------------------------------------
+#   |   rhel6   |           rhel.6-x64              |
+#   |  alpine*  |        linux-musl-(arch)          |
+#   |  freeBSD  |        freebsd.(version)-x64      |
+#
+# It is important to note that the function does not return anything, but it 
+# will set __DistroRid if there is a non-portable distro rid to be used.
+#
+initNonPortableDistroRid()
+{
+    # Make sure out parameter is cleared.
+    __DistroRid=
+
+    local buildOs=$1
+    local buildArch=$2
+    local isPortable=$3
+    local rootfsDir=$4
+
+    if [ "$buildOs" = "Linux" ]; then
+        if [ -e "${rootfsDir}/etc/redhat-release" ]; then
+            local redhatRelease=$(<${rootfsDir}/etc/redhat-release)
+
+            if [[ "${redhatRelease}" == "CentOS release 6."* || "$redhatRelease" == "Red Hat Enterprise Linux Server release 6."* ]]; then
+                nonPortableBuildID="rhel.6-${buildArch}"
+            fi
+        elif [ -e $rootfsDir/android_platform ]; then
+            source $rootfsDir/android_platform
+            nonPortableBuildID="$RID"
+        fi
+
+        # RHEL 6 is the only distro we will check redHat release for.
+        if [ -e "${rootfsDir}/etc/os-release" ] && [ "${nonPortableBuildID}" == "" ] ; then
```

Comment:
Last nit - it is not clear to me why we first check the /etc/redhat-release and /android_platform and then the /etc/os-release. If the /etc/os-release is present, it is the best source of truth, so it seems we should check that first and try the /etc/redhat-release and the /android_platform only if that file doesn't exist.

Reply:
Have made the change. Will verify the correct RIDs are selected.